server/mediaserver.go

```diff
@@ -395,21 +383,25 @@ func (s *LivepeerServer) registerConnection(rtmpStrm stream.RTMPVideoStream) (*r
 	if params == nil {
 		return nil, errMismatchedParams
 	}
-	mid := params.mid
+	mid := params.ManifestID
 	if drivers.NodeStorage == nil {
 		glog.Error("Missing node storage")
 		return nil, errStorage
 	}
-	storage := drivers.NodeStorage.NewSession(string(mid))
+	var storage drivers.OSSession
 	// Build the source video profile from the RTMP stream.
-	if params.resolution == "" {
-		params.resolution = fmt.Sprintf("%vx%v", rtmpStrm.Width(), rtmpStrm.Height())
+	if params.Resolution == "" {
+		params.Resolution = fmt.Sprintf("%vx%v", rtmpStrm.Width(), rtmpStrm.Height())
+	}
+	if params.OS == nil {
```

Comment:
Wouldn't this mean that if `params.OS` is not nil then a nil `storage` will be passed to `NewBasicPlaylistManager()` on L416 instead of `params.OS`?

Reply:
Well spotted. Fixed in c204e276def1ea4ed10188cdbec18107c755826e

Technically the storage is not used anymore within the playlist manager. Considered removing it from the playlist manager entirely, but opted to keep the scope of changes in this PR narrow.